### PR TITLE
Modify the crinit unit tests and crinit to support more fine-grained unit tests.

### PR DESCRIPTION
**Pull request by perotinus**
This does not add full test coverage for crinit, but it moves away from a monolithic test and starts implementing method-level tests. The intention is to add an integration test for the whole crinit flow, and test the intricacies of the logic in unit tests.

cc @madhusudancs, @font

### CODE REVIEW

File: pkg/crinit/init/init.go [hunk 1]
```diff
@@ -117,36 +113,55 @@ type initClusterRegistryOptions struct {
...
 	flags.StringVar(&o.etcdPVStorageClass, "etcd-pv-storage-class", "", "The storage class of the persistent volume claim used for etcd. Must be provided if a default storage class is not enabled for the host cluster.")
-	flags.BoolVar(&o.etcdPersistentStorage, "etcd-persistent-storage", true, "Use persistent volume for etcd. Defaults to 'true'.")
-	flags.BoolVar(&o.dryRun, "dry-run", false, "dry run without sending commands to server.")
-	flags.StringVar(&o.apiServerOverridesString, "apiserver-arg-overrides", "", "comma separated list of clusterregistry-apisererver arguments to override: Example \"--arg1=value1,--arg2=value2...\"")
+	flags.BoolVar(&o.etcdPersistentStorage, "etcd-persistent-storage", true, "Use a persistent volume for etcd. Defaults to 'true'.")
+	flags.BoolVar(&o.dryRun, "dry-run", false, "Run the command in dry-run mode, without making any server requests.")
+	flags.StringVar(&o.apiServerOverridesString, "apiserver-arg-overrides", "", "Comma-separated list of clusterregistry-apisererver arguments to override: Example \"--arg1=value1,--arg2=value2...\"")
```
Comment (madhusudancs) [inline pkg/crinit/init/init.go]: s/clusterregistry-apisererver/clusterregistry-apiserver/

Or even better, just write it as  "cluster registry API server", the way it is written in other places.

File: pkg/crinit/init/init_test.go [hunk 2]
```diff
@@ -18,1576 +18,491 @@ package init
...
-	if err := gcfg.ReadFileInto(&cfg, dnsProviderConfig); err != nil {
-		return nil, err
-	}
-
-	deployment.Annotations[util.FedDNSZoneName] = dnsZoneName
-	deployment.Annotations[util.FedNameServer] = cfg.Global.CoreDNSEndpoints
-	deployment.Annotations[util.FedDNSProvider] = util.FedDNSProviderCoreDNS
-	return deployment, nil
-}
+//func TestGenerateCredentials(t *testing.T) {
```
Comment (madhusudancs) [inline pkg/crinit/init/init_test.go]: Remove unused/incomplete code from here to line 508. You can always bring it back later?

File: pkg/crinit/init/init.go (hunk 1 above)
Comment (perotinus) [reply pkg/crinit/init/init.go]: That makes sense. Done.

File: pkg/crinit/init/init_test.go (hunk 2 above)
Comment (perotinus) [reply pkg/crinit/init/init_test.go]: Agreed. Removed.

**Pull request merged**